src/logger.rs

```diff
@@ -0,0 +1,100 @@
+use std::fs::File;
+use std::io::Write;
+use std::time::SystemTime;
+
+use anyhow::{Error, Result};
+use chrono::prelude::*;
+
+use crate::paths;
+
+pub struct Logger {
+    file: Option<File>,
+    error: Option<Error>,
+    level: i8,
+}
+
+impl Logger {
+    #[cfg_attr(feature = "cargo-clippy", allow(clippy::new_without_default))]
+    pub fn new() -> Logger {
+        let int_level = std::env::var("CYBERCMD_DEBUG")
+            .unwrap_or("1".to_string())
+            .parse::<i8>()
+            .unwrap_or(1);
+        let log_file_path = match paths::get_log_file_path() {
+            Ok(path) => path,
+            Err(e) => return Logger::from_error(e),
+        };
+
+        let is_modified_before_today = DateTime::<Local>::from(
+            std::fs::metadata(&log_file_path)
+                .ok()
+                .and_then(|md| md.modified().ok())
+                .unwrap_or(SystemTime::UNIX_EPOCH),
+        )
+        .date_naive()
+            < Local::now().date_naive();
+
+        let file = match File::options()
+            .create(true)
+            .write(true)
+            .truncate(is_modified_before_today)
+            .append(!is_modified_before_today)
+            .open(log_file_path)
+        {
+            Ok(file) => file,
+            Err(e) => return Logger::from_error(Error::from(e)),
+        };
+
+        let mut this = Logger {
+            file: Some(file),
+            error: None,
+            level: int_level,
+        };
+
+        this.debug("Starting cybercmd");
+
+        this
+    }
+
+    pub fn from_error(error: Error) -> Logger {
+        Logger {
+            file: None,
+            error: Some(error),
+            level: 0,
+        }
+    }
+
+    pub fn log(&mut self, level: i8, message: String) -> Result<()> {
```

Comment:
as mentioned above all log methods should probably accept `impl Display`, it's more convenient and doesn't force the caller to allocate strings - a value like `format_args!("write {0} to log", 1)` can be passed in without ever heap allocating

Reply:
I'm going to see if I can get one of the premade logging crates to work. Probably one that uses the log facade, which brings with it macros like `log::info!(..)`. I think these should fulfil your suggestions, giving `format!`-like macros that take traits like Display and Debug.